earth_enterprise/src/common/khMTTypes.h

```diff
@@ -119,6 +123,44 @@ class khMTQueue
 };
 
 
+class khRWLock
+{
+  private:
+    pthread_rwlock_t rwlock;
+  public:
+    khRWLock() : rwlock(PTHREAD_RWLOCK_INITIALIZER) { }
```

Comment:
I heard that `PTHREAD_RWLOCK_INITIALIZER` is only supposed to be used for statically allocated objects. It might be safer to call `pthread_rwlock_init` (or whatever the function name is) inside the constructor.